File: frontendEcovision/src/api/recommendation.js
import { apiClient, toFriendlyError } from './client'

// POST /recommendation { detected_object, confidence }
// -> { detected_object, confidence, container, explanation, recommendation }
export async function getRecommendation({ detectedObject, confidence }) {
  try {
    const { data } = await apiClient.post('/recommendation', {
      detected_object: detectedObject,
      confidence,
    })
    return { ok: true, data }
  } catch (error) {
    return { ok: false, error: toFriendlyError(error) }
  }
}

File: backend/app/schemas/recommendation.py
from pydantic import BaseModel


class RecommendationRequest(BaseModel):
    detected_object: str
    confidence: float


class RecommendationResponse(BaseModel):
    detected_object: str
    confidence: float
    container: str
    explanation: str
    recommendation: str
    prompt_tokens: int
    completion_tokens: int
    total_tokens: int
    api_cost: float